src/NuGet.Clients/PackageManagement.UI/DetailControlModel.cs

```diff
@@ -61,9 +61,14 @@ protected DetailControlModel(IEnumerable<NuGetProject> nugetProjects)
             OnPropertyChanged("IconUrl");
 
             var versions = await searchResultPackage.Versions.Value;
-
+                        
             _allPackageVersions = versions.Select(v => v.Version).ToList();
             CreateActions();
+            OnCurrentPackageChanged();
+        }
+
+        protected virtual void OnCurrentPackageChanged()
+        {
```

Comment:
This virtual method is added so that when this happens, in solution manager, we can update the applicable package manager providers for each project in the solution.